### PR TITLE
Feat: Portal Switch component

### PR DESCRIPTION
**Pull request by fayazara**
## Description

Adds a new component for switching portals in help center.

https://user-images.githubusercontent.com/15716057/178102130-cf195142-025b-4d72-9ce4-8246bf3bc0fc.mp4

Fixes https://github.com/chatwoot/product/issues/467

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue [hunk 1]
```diff
@@ -0,0 +1,194 @@
+<template>
+  <div class="portal" :class="{ active }">
+    <div class="avatar-container">
```
Comment (muhsin-k) [inline app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue:3]: There is no declaration for the class name `avatar-container`

Comment (fayazara) [reply app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue]: Fixed @muhsin-k

**Review by muhsin-k — DISMISSED** [review]

File: app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue [hunk 2]
```diff
@@ -0,0 +1,164 @@
+<template>
+  <div class="portal" :class="{ active }">
+    <thumbnail :username="portal.name" variant="square" />
+    <div class="actions-container">
+      <header>
+        <div>
+          <p class="portal-title">{{ portal.name }}</p>
```
Comment (nithindavid) [inline app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue:7]: Please use H level tags

File: app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+            {{ portal.article_count }}
+            {{ $t('HELP_CENTER.PORTAL.ARTICLES_LABEL') }}
+          </p>
+        </div>
+        <span v-if="active" class="badge">{{
+          $t('HELP_CENTER.PORTAL.ACTIVE_BADGE')
+        }}</span>
+      </header>
+      <div class="portal-locales">
+        <p class="locale-title">
```
Comment (nithindavid) [inline app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue:18]: Use h level tags here too

File: app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  mounted() {
+    this.selectedLocale = this.portal.locales[0].code;
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+.portal {
+  background-color: var(--white);
+  padding: var(--space-one);
```
Comment (nithindavid) [inline app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue:80]: Can we use exact padding as per design?
```suggestion
  padding: var(--space-normal);
```

File: app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+  border: 1px solid var(--color-border-dark);
+  border-radius: var(--border-radius-medium);
+  position: relative;
+  display: flex;
+  margin-bottom: var(--space-one);
```
Comment (nithindavid) [inline app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue:85]: We usually try to use powers of 2 for spacing `(2, 4, 8 , 16)`, And uses the `one, slab` when it's unavoidable. 

I think we can use `small or normal (8, 16)`  here

File: app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+    margin-left: var(--space-one);
+    flex-grow: 1;
+
+    header {
+      display: flex;
+      align-items: flex-start;
+      justify-content: space-between;
+      margin-bottom: var(--space-one);
+
+      .badge {
```
Comment (nithindavid) [inline app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue:101]: We can use existing label component with `smooth` here.

File: app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+        font-size: var(--font-size-mini);
+        background-color: var(--w-100);
+        color: var(--w-600);
+        padding: var(--space-smaller) var(--space-small);
+      }
+
+      .portal-title {
+        color: var(--s-700);
```
Comment (nithindavid) [inline app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue:109]: Can we use correct color as per design.

```suggestion
        color: var(--s-900);
```

File: app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+        font-size: var(--font-size-default);
```
Comment (nithindavid) [inline app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue:110]:
```suggestion
        font-size: var(--font-size-medium);
```

**Review by nithindavid — CHANGES_REQUESTED** [review]
Minor design comments left

File: app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue (hunk 8 above)
Comment (fayazara) [reply app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue]: FIxes

File: app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue (hunk 6 above)
Comment (fayazara) [reply app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue:101]: Screenshot2022-07-18 at 00 45 44@2x

@nithindavid  I tried using class="badge smooth" but I dont think we have this variant of the badge, hence wrote a custom class for this to match the design.

**Review by iamsivin — COMMENTED** [review]
Please see the minor comments

File: app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue (hunk 6 above)
Comment (nithindavid) [reply app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue:101]: Works for now

File: app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue [hunk 9]
```diff
@@ -0,0 +1,164 @@
+<template>
+  <div class="portal" :class="{ active }">
+    <thumbnail :username="portal.name" variant="square" />
+    <div class="actions-container">
+      <header>
+        <div>
+          <h2 class="portal-title">{{ portal.name }}</h2>
```
[the remark this answers is not in the document]
Comment (fayazara) [reply app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue]: Resolved @iamsivin

**Review by muhsin-k — APPROVED** [review]
@fayazara Please fix the conflicts

**Review by iamsivin — APPROVED** [review]

File: app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+  position: relative;
+  display: flex;
+  margin-bottom: var(--space-normal);
+
+  &.active {
+    border: 1px solid var(--w-400);
+  }
+
+  .actions-container {
+    margin-left: var(--space-one);
```
Comment (nithindavid) [inline app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue:95]: Use `small` here

File: app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+
+      .badge {
+        font-size: var(--font-size-mini);
+        background-color: var(--w-100);
+        color: var(--w-600);
+        padding: var(--space-smaller) var(--space-small);
+      }
+
+      .portal-title {
+        color: var(--s-900);
```
Comment (nithindavid) [inline app/javascript/dashboard/components/helpCenter/Portal/Switch/Index.vue:112]: We can remove these styles, color will be there by default, for font-weight we could stick to `medium` as per design.

**Review by nithindavid — CHANGES_REQUESTED** [review]
Few minor more and we are good to merge. 🚀

**Review by muhsin-k — DISMISSED** [review]

**Review by nithindavid — APPROVED** [review]

File: app/javascript/dashboard/i18n/locale/en/helpCenter.json [hunk 12]
```diff
@@ -26,6 +26,13 @@
       "SAVING": "Draft saving...",
       "SAVED": "Draft saved"
     },
+    "PORTAL": {
+      "ACTIVE_BADGE": "active",
+      "CHOOSE_LOCALE_LABEL": "Choose a locale",
+      "ARTICLES_LABEL": "articles",
+      "ADD_NEW_LOCALE": "Add a new locale"
+    }
```
Comment (iamsivin) [inline app/javascript/dashboard/i18n/locale/en/helpCenter.json:34]:
```suggestion
    },
```

File: app/javascript/dashboard/i18n/locale/en/helpCenter.json [hunk 13]
```diff
@@ -26,6 +26,13 @@
...
+  },
```
Comment (iamsivin) [inline app/javascript/dashboard/i18n/locale/en/helpCenter.json:35]:
```suggestion

```

File: app/javascript/dashboard/i18n/locale/en/helpCenter.json [hunk 14]
```diff
@@ -79,5 +86,4 @@
         }
       }
     }
-  }
```
Comment (iamsivin) [inline app/javascript/dashboard/i18n/locale/en/helpCenter.json:82]: @fayazara Revert this changes

**Review by iamsivin — COMMENTED** [review]
Please see these comments.

**Review by muhsin-k — APPROVED** [review]

**Review by iamsivin — APPROVED** [review]
